code/modules/atmospherics/machinery/components/unary_devices/vent_pump.dm

```diff
@@ -153,6 +153,18 @@
 	else // pump_direction == SIPHONING
 		icon_state = "vent_in"
 
+/obj/machinery/atmospherics/components/unary/vent_pump/should_atmos_process(datum/gas_mixture/air, exposed_temperature)
+	if(!is_operational)
+		return FALSE
+	if(!nodes[1])
+		on = FALSE
```

Comment:
I don't like that a proc like `should_XXX` is performing side effects. @LemonInTheDark is this normal for atmos code?

Reply:
no this is born from confusingly named procs